build-setup/pytorch/setup_pytorch.md

```diff
@@ -0,0 +1,42 @@
+# Setup for Pytorch
+
+If you wish to build and run the PyTorch inference applications you will need to install Pytorch and build the torch libraries. 
```

Comment:
The language was intentional this is an optional step for everyone that is the reason the top level makefile does not build the the pytorch binary. If the future once this code is in production I can still see the value of having this step optional, if you just want to build anomaly detection why do you have to install anaconda, clone the PyTorch repo etc. 
The use of a separate file in a subfolder is following the pattern set by `clion` and `strptime` and because the initial setup steps are the same for each platform.

Reply:
It's only an optional step while this is a dev-only prototype.  If we are ever going to ship this then it should be in the top level per-OS files.

`clion` is truly optional even for shipped code.  It's not installed on any of the CI machines.

`strptime` has a separate directory because the patches are too complex to inline in the top level build setup instructions.  But the instructions for using the files in that directory _are_ in the top level build setup instructions [here](https://github.com/elastic/ml-cpp/blob/master/build-setup/windows.md#strptime).

You could say "the initial setup steps are the same for each platform" for Boost.  So instead of having a Boost section in `macos.md`, `linux.md` and `windows.md` we could say in each of these files "now build Boost using the instructions in `boost/boost.md` and direct people to another file that has separate sections for macOS, Linux and Windows.  But I am not sure that would really make life easier for someone who had been tasked with setting up a particular type of build server.  They'd have to look at two files instead of one.

So the instructions could stay as-is for this PR, while we're not actually building the new program on every build, but eventually they need to move to be consistent with the other instructions.